samples/subsys/senss/src/main.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <senss/senss.h>
+#include <stdio.h>
+
+static int data_event_callback(
+		senss_sensor_handle_t handle,
+		void *buf, int size, void *param)
+{
+	printf("Received sensor data from sensor: %s, type: %d, instance: %d\n",
+		SENSS_SENSOR_INFO_NAME(handle),
+		SENSS_SENSOR_INFO_SENSOR_TYPE(handle),
+		SENSS_SENSOR_INFO_INSTANCE(handle)
+	);
+
+	int type = SENSS_SENSOR_INFO_SENSOR_TYPE(handle);
+	if (type == SENSS_SENSOR_TYPE_MOTION_ACCELEROMETER_3D) {
+		struct senss_sensor_value_3d_int32* value = buf;
+		printf("Received sensor data readings :%d\n",
+			value->header.reading_count);
+		for (int i = 0; i < value->header.reading_count; i++) {
+			printf("Accel sensor data : x: %d, y: %d, z: %d\n",
+				value->readings[i].x,
+				value->readings[i].y,
+				value->readings[i].z);
+		}
+	}
+}
+
+void main(void)
+{
+	int ret;
+	ret = senss_init();
+	if (!ret) {
+		printf("Failed to init sensor subsystem! \n");
+		return;
+	}
+
+	/* Enumerate all sensor instances */
+	int sensor_count;
+	sensor_count = senss_get_sensor_count(SENSS_SENSOR_TYPE_ALL, -1);
```

Comment:
Does this build? I'm not seeing an implementation of `senss_get_sensor_count()` in this PR

Reply:
No,  its not build,  at current there is no any implementations yet,  this code is just a example for how to use the `sensor subsystem` API. 

But it can pass compile stage, and has problem in link stage due to linked unimplemented symbols,  I also used it for common syntax checking for the API headers. To avoid basic common errors.  :)